networkx/algorithms/traversal/breadth_first_search.py

```diff
@@ -58,10 +62,16 @@ def generic_bfs_edges(G, source, neighbors=None, depth_limit=None):
     .. _PADS: http://www.ics.uci.edu/~eppstein/PADS/BFS.py
     .. _Depth-limited-search: https://en.wikipedia.org/wiki/Depth-limited_search
     """
+    def get_neighbors(node):
```

Comment:
How about: 
```python
if callable(sort_neighbors):
    neighbors = lambda node: sort_neighbors(neighbors(node))
```
Then you don't have to use get_neighbors and avoid excess processing? Are their other effects?

Reply:
I think using `lambda` function will cause a `RecursionError`, thats why I used a regular functions.

Reply:
The lambda function won't cause a RecursionError because it doesn't use recursion on itself. It is inside a recursive function, but the lambda itself is not recursive. I'm pretty sure that will be OK.  

You could also use regular functions if you prefer with something similar to (lots of ways to do it):
```python
if callable(sort_neighbors):
    def get_neighbors(node):
        return sort_neighbors(neighbors(node))
else:
    get_neighbors = neighbors
```

Reply:
when calling a `lambda`  the `neighbors` inside will evaluate to the defined `lambda` (which have the same name `neighbors`) , so there will be a recursion error, minimal example:

```
    def adjecent(x):
      return (x - 1, x + 1)

    adjecent = lambda x: sorted(adjecent(x))

    print(adjecent(10))
```

will Output `RecursionError: maximum recursion depth exceeded`. But I found a way to do it:

```
    if callable(sort_neighbors):
        _neighbors = neighbors
        neighbors = lambda node: iter(sort_neighbors(_neighbors(node)))
```

The `iter` is needed because the `sort_neighbors` may return an iterable instead of an iterator. I will update to this implementation , if you OK with it?

Reply:
That's great -- Thanks!